src/sentry_transport.c

```diff
@@ -165,9 +167,24 @@ sentry__prepare_http_request(sentry_envelope_t *envelope,
         return NULL;
     }
 
+    char *compressed_body = NULL;
+    size_t compressed_body_len = 0;
+    bool compressed = sentry__gzipped_with_compression(
+        body, body_len, &compressed_body, &compressed_body_len);
+    if (compressed) {
+        if (body_owned) {
+            sentry_free(body);
+            body = NULL;
+            body_owned = false;
+        }
```

Comment:
I rather mean this:
```suggestion
        if (body_owned) {
            sentry_free(body);
        }
        body = compressed_body;
        body_len = compressed_body_len;
        body_owned = true;
```

That way, you do not need to duplicate all the code below depending on `if (compressed)`, as everything will just work the same way with `body`.

Reply:
It does, as the `compressed_body` is definitely owned and needs to be freed once you are re-assigning `body` here.

Reply:
Yes. I had modified. Thx